File: internal/model/analysis.go
package model

import (
	"time"
)

// Analysis 分析历史模型
type Analysis struct {
	ID            string    `json:"id" gorm:"primaryKey"`
	SessionID     string    `json:"session_id" gorm:"index"`              // 会话ID（可选）
	ResultsData   string    `json:"results_data" gorm:"type:text"`        // 执行结果数据（JSON格式）
	AnalysisResult string   `json:"analysis_result" gorm:"type:text"`     // 分析结果
	Question      string    `json:"question" gorm:"type:text"`            // 用户提问（可选）
	CreatedAt     time.Time `json:"created_at"`
}

File: web/src/api/analysis.ts
import { request } from './request'

export interface AnalyzeRequest {
  results: unknown[]
  question?: string
}

export interface AnalyzeResponse {
  id: string
  analysis_result: string
  question?: string
  created_at: string
}

export interface Analysis {
  id: string
  session_id?: string
  results_data: string
  analysis_result: string
  question?: string
  created_at: string
}

export interface AnalysisHistoryParams {
  session_id?: string
  limit?: number
}

// ============================================================================
// Analysis API Functions
// ============================================================================

export function analyzeResults(data: AnalyzeRequest): Promise<AnalyzeResponse> {
  return request.post<AnalyzeResponse>('/analysis/analyze', data)
}

export function getAnalysisHistory(params?: AnalysisHistoryParams): Promise<Analysis[]> {
  return request.get<Analysis[]>('/analysis/history', { params })
}

export function getAnalysis(id: string): Promise<Analysis> {
  return request.get<Analysis>(`/analysis/${id}`)
}

export function deleteAnalysis(id: string): Promise<void> {
  return request.delete(`/analysis/${id}`)
}
